Dockerfile

```diff
@@ -2,17 +2,23 @@ FROM python:3.11-slim
 
 WORKDIR /app
 COPY ./requirements.txt /app
+
+ENV BUILD_DEPS="libssl-dev cargo gcc g++ libffi-dev"
+
 # "Installing this module requires OpenSSL python bindings"
-RUN BUILD_DEPS="libssl-dev cargo gcc libffi-dev" && \
-    apt-get update && apt-get install -y $BUILD_DEPS && \
+RUN apt-get update && apt-get install -y $BUILD_DEPS && \
     PYOPENSSL=$(grep 'pyopenssl=' requirements.txt) && \
     pip install --no-cache-dir $PYOPENSSL && \
-    pip install --no-cache-dir -r /app/requirements.txt && \
+    pip install --no-cache-dir -r /app/requirements.txt
+
+COPY . /app
+
+RUN cd /app/distance && g++ -std=c++14 -o /usr/local/bin/distance distance.cpp && \
     apt-get purge -y --auto-remove $BUILD_DEPS && \
```

Comment:
this has to stay on the upper RUN. otherwise the packages will get installed on the layer on Line 9, and be removed on this layer. so you download them to add them and remove them.

Reply:
Ok. Makes sense